core/client/fs/src/test/java/alluxio/client/block/AlluxioBlockStoreTest.java

```diff
@@ -264,6 +265,32 @@ public void getInStreamUfs() throws Exception {
     assertEquals(worker2, mBlockStore.getInStream(BLOCK_ID, options).getAddress());
   }
 
+  @Test
+  public void getInStreamNoWorkers() throws Exception {
+    URIStatus dummyStatus =
+        new URIStatus(new FileInfo().setPersisted(true).setBlockIds(Collections.singletonList(0L)));
+    InStreamOptions options = new InStreamOptions(dummyStatus, OpenFileOptions.defaults());
+    when(mMasterClient.getBlockInfo(BLOCK_ID)).thenReturn(new BlockInfo());
+    when(mMasterClient.getWorkerInfoList()).thenReturn(Collections.emptyList());
+
+    mException.expect(UnavailableException.class);
+    mException.expectMessage("No Alluxio worker available");
+    mBlockStore.getInStream(BLOCK_ID, options).getAddress();
+  }
+
+  @Test
+  public void getInStreamMissingBlock() throws Exception {
+    URIStatus dummyStatus = new URIStatus(
+        new FileInfo().setPersisted(false).setBlockIds(Collections.singletonList(0L)));
+    InStreamOptions options = new InStreamOptions(dummyStatus, OpenFileOptions.defaults());
+    when(mMasterClient.getBlockInfo(BLOCK_ID)).thenReturn(new BlockInfo());
+    when(mMasterClient.getWorkerInfoList()).thenReturn(Collections.emptyList());
+
+    mException.expect(NotFoundException.class);
+    mException.expectMessage("unavailable in both Alluxio and UFS");
```

Comment:
Use the exception message template?

Reply:
testing the template could lead to issues where the template changes and stops being relevant to the case being tested